docs/en/quiz/chapter02.md

```diff
@@ -198,7 +198,7 @@ def α : Prod (Type 0) (Type 1) := (Nat, Type)
 \(b\) `Type 2 → Type 3`
 ```lean
 def q09d : Type 2 -> Type 3 :=
-  fun x : Type 2 => x
+  fun x : Type 2 => x -> Type 2
```

Comment:
제가 처음에 이렇게 푼 이유는 린의 유니버스 레벨을 고려했을 때 Type 3은 Type 2를 타입으로 가지고 있기 때문에(Type 2가 Type 3에 속하기 때문에) x를 그대로 반환하는 쪽으로 생각했습니다. 
그런데 멘토님께서 바로 잡아주신 코드를 보고 제가 기존에 작성했던 `fun x : Type 2 => x`는 x를 그대로 반환하여 Type 2에 속하는데 해당 함수가 Type 3을 반환해야한다는 점에서 조건을 충족하지 못했기 때문에 오답임을 알게 되었습니다. 
린에서 함수 타입의 유니버스 레벨(max u v) 관점을 고려했을 때 x는 Type 2이므로  u=2, Type 2는 Type 3에 속하므로 반환 타입의 레벨은 v=3이라서 x -> Type 2의 타입은 Type 3이 되는 방식을 놓쳤던 것 같습니다. 감사합니다!

Reply:
> 린에서 함수 타입의 유니버스 레벨(max u v) 관점을 고려했을 때 x는 Type 2이므로 u=2, Type 2는 Type 3에 속하므로 반환 타입의 레벨은 v=3이라서 x -> Type 2의 타입은 Type 3이 되는 방식을 놓쳤던 것 같습니다. 감사합니다!

항 `x`의 유형이 `Type 2`일 때, 식 `x -> Type 2`의 유형이 `Type 3`인 이유는 그 유형이 `Type (max 2 3)`이기 때문이 아니라, 그 유형이 `Type (imax 2 3)`이기 때문입니다. 의존 함수형의 유형 부여 규칙은 "린 4로 하는 정리 증명"의 4장에야 나오는데, 그 내용은 다음과 같습니다.

[English translation] Given `x : Type 2`, the function type `x -> Type 2` has type `Type 3` not because its type is `Type (max 2 3)`, but because its type is `Type (imax 2 3)`. See the following discussion in Chapter 4 of "Theorem Proving in Lean 4":

From https://docs.lean-lang.org/theorem_proving_in_lean4/quantifiers_and_equality.html:
> It is the typing rule for dependent arrow types, and the universal quantifier in particular, that distinguishes `Prop` from other types. Suppose we have `α : Sort i` and `β : Sort j`, where the expression `β` may depend on a variable `x : α`. Then `(x : α) → β` is an element of `Sort (imax i j)`, where `imax i j` is the maximum of `i` and `j` if `j` is not 0, and 0 otherwise.

Reply:
넵! 의존 함수형의 유형 부여 규칙에 대한 내용도 교재 4장에서 살펴보도록 하겠습니다.